lib/helpers/configuration.rb

```diff
@@ -20,10 +20,11 @@ def configuration
     yield self
   end
 
-  def define_setting(name, default = nil)
+  def define_setting(name, default = nil, allow_nil: true)
     class_variable_set("@@#{name}", default)
 
     define_class_method "#{name}=" do |value|
+      raise "#{name} cannot be nil." if value.nil? && !allow_nil
```

Comment:
Agreed! Will fix these issues.

In `README.md` you suggest creating an initializer named `share_meow.rb` whereas the classname is `ShareMeowClient`. Is this on purpose? I'd say `share_meow_client.rb` might be  better. The reason I ask is because in your suggested error message you refer to the "ShareMeow initializer" whereas I'd say "ShareMeowClient initializer".